### PR TITLE
Switch from OC::$server->get to OCP\Server::get

### PR DESCRIPTION
**Pull request by CarlSchwan**
* Target version: master 

### Summary

Switch from OC::$server->get to OCP\Server::get

And add a bit more typing to some classes

### TODO

- [ ] Make the test works

### Checklist

- [x] Code is properly formatted
- [x] Sign-off message is added to all commits
- [x] Tests (unit, integration, api and/or acceptance) are included
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by CarlSchwan** [comment]
Ended up fixing a few more psalm issues :laughing:

**Comment by juliushaertl** [comment]
🤩 Very nice, also with the stub for circles

**Comment by CarlSchwan** [comment]
The remaining integration failure are looking to be related to server change

Undefined mixin icon-black-white: apps\/deck\/css\/deck.scss This needs to be ported away from scss compiled by the server

**Comment by juliushaertl** [comment]
Fixes for icon issues are in https://github.com/nextcloud/deck/pull/3803

**Comment by juliushaertl** [comment]
Pushed a fix for the tests to use the SymfonyAdapter in the template and opened https://github.com/nextcloud/server/pull/32349 to get rid of that one.

File: lib/Service/ConfigService.php [hunk 1]
```diff
@@ -40,23 +40,26 @@ class ConfigService {
...
-	private $userId;
-	private $groupManager;
+	private IConfig $config;
+	private ?string $userId = null;
+	private IGroupManager $groupManager;
 
 	public function __construct(
 		IConfig $config,
-		IGroupManager $groupManager
+		IGroupManager $groupManager,
```
Comment (CarlSchwan) [inline lib/Service/ConfigService.php:49]:
```suggestion
		IGroupManager $groupManager
```

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**